examples/asa/assets/5-contract-owned-asa-approval.py

```diff
@@ -0,0 +1,80 @@
+from pyteal import *
+
+def contract_owned_asa():
+    '''
+    This program is stateful part of contract owned ASA, This program keeps track of
+    number of ASA created by the contract, It also checks for
+    : While payment only creator can receive the payments
+    : While asset creation only creator can create assets
+    Owner can also be changed using this program
+    '''
+
+    on_creation = Seq([
+        App.globalPut(Bytes("Creator"), Txn.sender()),
+        App.globalPut(Bytes("ASACounter"), Int(0)),
+        Return(Int(1))
+    ])
+
+    # verify neither transaction
+    # contains a rekey
+    rekey_check = And(
+        Gtxn[0].rekey_to() == Global.zero_address(),
+        Gtxn[1].rekey_to() == Global.zero_address()
+    )
+
+    on_asa_create = Seq([
+        Assert(And(
+            App.globalGet(Bytes("Creator")) == Txn.sender(),
+            App.globalGet(Bytes("ASACounter")) == Int(0), # limit no of asa from here: currently set to max 1 ASA.
+            Global.group_size() == Int(3),
+            Gtxn[1].type_enum() == TxnType.AssetConfig,
+            Gtxn[2].type_enum() == TxnType.Payment,
+            rekey_check,
+            Gtxn[2].rekey_to() == Global.zero_address(),
+            Gtxn[2].amount() == Int(1000000)
+        )),
+        App.globalPut(Bytes("ASACounter"), Int(1)),
```

Comment:
If we are keeping a counter here, then shouldn't we do counter++? 
`App.globalPut(Bytes("ASACounter"), App.globalGet(Bytes("ASACounter")) + Int(1))`

Reply:
No, for max 1 Asa, both works

Reply:
Why do we call it a counter if it will always be max 1 Asa? Say if user wants max 2 asa, this will again update counter to 1, on 2nd asa deploy

Reply:
Why do we call it a counter if it will always be max 1 Asa? Say if user wants max 2 asa, this will again update counter to 1, on 2nd asa deploy

Reply:
then user can update this counter, but this example is for 1 ASA. but can update if you want

Reply:
I think it's better to use the suggested code. Currently it looks ambigous